packages/cache/src/index.ts

```diff
@@ -0,0 +1,180 @@
+    } else {
+      throw new Error(`Unknown cache type "${this.type}"`);
+    }
+  }
+
+  set(key: string, value: any, maxAgeMS: number) {
+    if (!this.enabled) return;
+
+    const scopedKey = this.keyPrefix + key;
+
+    switch (this.type) {
+      case 'memory': {
+        assert(this.memoryCache, 'Memory cache is enabled but not configured');
+        this.memoryCache.set(scopedKey, JSON.stringify(value), { ttl: maxAgeMS });
+        break;
+      }
+
+      case 'redis': {
+        // This returns a promise, but we don't want to wait for this data
+        // to reach the cache before continuing, and we don't *really*
+        // care if it errors.
+        //
+        // We don't log the error because it contains the cached value,
+        // which can be huge and which fills up the logs.
+        assert(this.redisClient, 'Redis client is enabled but not configured');
+        this.redisClient
+          .set(scopedKey, JSON.stringify(value), 'PX', maxAgeMS)
+          .catch((_err) => logger.error('Cache set error', { key, scopedKey, maxAgeMS }));
+        break;
+      }
+    }
+  }
+
+  async del(key: string) {
+    if (!this.enabled) return;
+
+    const scopedKey = this.keyPrefix + key;
+
+    switch (this.type) {
+      case 'memory': {
+        assert(this.memoryCache, 'Memory cache is enabled but not configured');
+        this.memoryCache.delete(scopedKey);
+        break;
+      }
+
+      case 'redis': {
+        assert(this.redisClient, 'Redis client is enabled but not configured');
+        await this.redisClient.del(scopedKey);
+        break;
+      }
+    }
+  }
+
+  /**
+   * Returns the value for the corresponding key if it exists in the cache; null otherwise.
+   */
+  async get(key: string): Promise<any> {
+    if (!this.enabled) return null;
+
+    const scopedKey = this.keyPrefix + key;
+
+    switch (this.type) {
+      case 'memory': {
+        assert(this.memoryCache, 'Memory cache is enabled but not configured');
+        const value = this.memoryCache.get(scopedKey);
+        if (typeof value === 'string') {
+          return JSON.parse(value);
+        }
+        return undefined;
+      }
+
+      case 'redis': {
+        assert(this.redisClient, 'Redis client is enabled but not configured');
+        const value = await this.redisClient.get(scopedKey);
+        if (typeof value === 'string') {
+          return JSON.parse(value);
+        }
+        return undefined;
+      }
+
+      default: {
+        return null;
+      }
+    }
+  }
+
+  /**
+   * Clear all entries from the cache.
+   */
+  async reset() {
+    if (!this.enabled) return;
+
+    switch (this.type) {
+      case 'memory': {
+        assert(this.memoryCache, 'Memory cache is enabled but not configured');
+        this.memoryCache.clear();
+        break;
+      }
+
+      case 'redis': {
+        let cursor = '0';
+        do {
+          if (!this.redisClient) {
+            throw new Error('Redis client not initialized');
+          }
```

Comment:
Nit, for consistency with other checks:

```suggestion
          assert(this.redisClient, 'Redis client is enabled but not configured');
```

Reply:
Updated locally in [120c12b](https://github.com/PrairieLearn/PrairieLearn/pull/9360/commits/120c12b2b370c18d126ea7b451843f8ac7ff2f8a).